fix(preset:angular): escaped scoped string should not be seen as GitHub username (#601)

SuperITMan opened the pull request:
ISSUES CLOSED: #601

SuperITMan committed:
fix(preset:angular): escaped scoped string should not be seen as GitHub username (#601)

SuperITMan commented:
@dudekaa @bcoe could you please have a look at this? 😊

dudekaa commented:
Hi @SuperITMan 

thanks for contributing...but as first note: I'm no more attached to this project than you, so for valid answers you have to ask project maintainers. I can make only advices here!

Now, I feel you' re patching this at wrong place at all, so it gets easily over-comlicated... as I understand your bug report, you simply want to teach this RegExp
```javascript
commit.subject = commit.subject.replace(/\B@([a-z0-9](?:-?[a-z0-9/]){0,38})/g, (_, username) => {
```
to simply don't match quoted user refs and you should be fine... It' s much more simpler and understandable approach...

But once more again, I' m just contributor as you, so you can't take me for granit.

BR
Arnost

bcoe commented:
@SuperITMan sorry that we dropped this PR on the floor. @dudekaa I agree that I think we could figure out a slight rework of `commit.subject.replace` and address the problem.

SuperITMan commented:
Hello @bcoe @dudekaa 
Thanks for the remark, I'll have a look in the next days to improve this :-)

SuperITMan commented:
I updated a bit my PR and added some comments to explain why the check is there.
With simple regex, there is unfortunately no way to check the number of occurrences of a character (in this case: `).

If you have a better solution in mind, please, feel free to share it :blush:

SuperITMan commented:
@bcoe I updated my PR. Could you have a look at this ?

bcoe commented:
@SuperITMan thanks for updating the PR 👍 will try to help merge some of these PRs soon.
